metaflow/plugins/kubernetes/kubernetes_job.py

```diff
@@ -339,15 +339,16 @@ def __init__(self, client, name, uid, namespace):
         self._job = self._fetch_job()
         self._pod = self._fetch_pod()
 
-        import atexit
+        import signal
```

Comment:
what's the motivation for this change?

Reply:
Without this change, the code would try to fetch the job object and then patch it to set parallelism to 0. This even when knowing that the container has actually completed execution. So, it's only a matter of time when the job is also marked completed. Having the original code using `atexit` as is reduces the latency improvements introduced in this change.

AFAICT, the original goal of this code was to cleanup the pods (best-effort) in the event of the user-created process running Metaflow is terminated. The change continues to do that.

Let me know if there was any additional motivation for having this and I can test that scenario out.

Reply:
`atexit` was also supposed to handle [this scenario](https://github.com/Netflix/metaflow/blob/master/metaflow/plugins/kubernetes/kubernetes_cli.py#L239) - which would break after this change.